### ISSUE TITLE
[Rust] Use planus as a cargo build script?

### ISSUE THREAD

**Issue by TheButlah** [planus-org/planus#128]
Would be cool to allow writing a flatbuffer schema, and then using planus in a cargo build script to generate the rust code from the schema as a build step, automagically. This would make the ergonomics of the flatbuffer code just as nice as any other cargo dependency.

If this isn't currently supported, is there any interest in accepting a PR for this? I'm guessing its low hanging fruit.

**Comment by TethysSvensson** [comment]
This sounds like a good idea. It's somewhat related to #18, though that issue is a bit bigger. I think we should be able to solve this issue first and then do further re-organizing later to support #18.

**Comment by mbodmer** [comment]
I am not sure whether this is a new function, but what about?

```toml
[build-dependencies]
planus-codegen = "1.0.0"
planus-translation = "1.0.0"
```

```rust
use std::io::Result;

fn main() -> Result<()> {
    let out_dir = std::env::var("OUT_DIR").unwrap();

    let Some(declarations) = planus_translation::translate_files(&["fbs/monster.fbs"]) else {
        panic!("Cannot translate code for fbs/monster.fbs")
    };
    let code = planus_codegen::generate_rust(&declarations)
        .expect("Cannot generate code for fbs/monster.fbs");
    std::fs::write(format!("{out_dir}/monster.rs"), code)
        .expect("Cannot write generated fbs: {out_dir}/monster.rs");

    Ok(())
}
```

**Comment by TethysSvensson** [comment]
In my previous comment I mentioned #18, which was solved in #161. I think you proposal works well.
